### PR TITLE
Feat filter

### PR DESCRIPTION
**Pull request by Kothalkarpriya**
added filters to the product site. It is not completely working but added main functions to it

### CODE REVIEW

File: src/components/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                type="radio"
+                value="PRICE_A"
+                id="sortByPrice"
+                onClick={(e) =>
+                  filterDispatch({ type: "SORT", payload: e.target.value })
+                }
+              />
+              <label htmlFor="sortByPrice">Price(Low to High)</label>
+            </li>
+            <li className="filter-list-item">
```
Comment (mariyasada) [inline src/components/Filter.jsx]: don't use PRICE_A ,use something like price_sortBYAcending

File: src/components/ProdCard.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { AiTwotoneHeart } from "react-icons/ai";
+// import { BsStar } from "react-icons/bs";
+
+export default function ProdCard({ cardTitle, cardRating, cardPrice, image }) {
+  // const rounded = Math.round(cardRating * 2) / 2;
+  // const hasDecimal = rounded % 1 !== 0;
+  // const roundedWhole = Math.floor(rounded);
+  // const stars = hasDecimal ? roundedWhole + 1 : roundedWhole;
+  return (
```
Comment (mariyasada) [inline src/components/ProdCard.jsx]: remove commented code

File: src/components/ProdCard.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        </div>
+        {/* <div>
+          {[...Array(roundedWhole)].map((e, i) => (
+            <BsStar key={i} />
+          ))}
+          {hasDecimal ? <i class="fas fa-star-half-alt"></i> : ``}
+          {[...Array(5 - stars)].map((e, i) => (
+            <BsStar key={i} />
+          ))}
+        </div> */}
```
Comment (mariyasada) [inline src/components/ProdCard.jsx]: here also

File: src/context/prodContext.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+function ProdContext({ children }) {
+  return (
+    <ProductContext.Provider value={{ products }}>
+      {children}
+    </ProductContext.Provider>
+  );
+}
+
+const GetProd = () => useContext(ProductContext);
+
```
Comment (mariyasada) [inline src/context/prodContext.jsx:14]: write full variable name getProduct instead of GetProd

**Review by mariyasada — APPROVED** [review]
everything looks good. just added some suggestions

File: src/components/Filter.jsx (hunk 1 above)
Comment (Kothalkarpriya) [reply src/components/Filter.jsx]: ok I will change

File: src/components/ProdCard.jsx (hunk 3 above)
Comment (Kothalkarpriya) [reply src/components/ProdCard.jsx]: yes I will

File: src/components/Filter.jsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            </li>
+          </ul>
+        </div>
+        <div className="prod-sorting">
+          <div className="filter-heading">Category</div>
+          <ul className="filter-list">
+            <li className="filter-list-item">
+              <input
+                name="sort"
+                checked={state.sort === "PRICE_A"}
```
Comment (ankit9015) [inline src/components/Filter.jsx]: Use more descriptive state values, PRICE_A can be improved

**Review by ankit9015 — APPROVED** [review]
The code looks good to merge, just go through the suggestions before merge

File: src/components/Filter.jsx (hunk 5 above)
Comment (Kothalkarpriya) [reply src/components/Filter.jsx]: Ok I will

**Pull request merged**